quantum/quantum.c

```diff
@@ -217,14 +217,82 @@ void reset_keyboard(void) {
   #define SFTENT_KEY KC_ENT
 #endif
 
-static bool shift_interrupted[2] = {0, 0};
-static uint16_t scs_timer[2] = {0, 0};
+// Control / paren setup
+
+#ifndef LCPO_KEY
+  #define LCPO_KEY KC_9
+#endif
+#ifndef RCPC_KEY
+  #define RCPC_KEY KC_0
+#endif
+
+#ifndef LCPO_MOD
+  #define LCPO_MOD KC_LCTL
+#endif
+#ifndef RCPC_MOD
+  #define RCPC_MOD KC_RCTL
+#endif
+
+// Alt / paren setup
+
+#ifndef LAPO_KEY
+  #define LAPO_KEY KC_9
+#endif
+#ifndef RAPC_KEY
+  #define RAPC_KEY KC_0
+#endif
+
+#ifndef LAPO_MOD
+  #define LAPO_MOD KC_LALT
+#endif
+#ifndef RAPC_MOD
+  #define RAPC_MOD KC_RALT
+#endif
+
+static bool space_cadet_interrupted[4] = {0, 0, 0, 0};
+static uint16_t space_cadet_timer[4] = {0, 0, 0, 0};
 
 /* true if the last press of GRAVE_ESC was shifted (i.e. GUI or SHIFT were pressed), false otherwise.
  * Used to ensure that the correct keycode is released if the key is released.
  */
 static bool grave_esc_was_shifted = false;
 
+static void process_space_cadet(keyrecord_t *record, uint8_t index, uint8_t normalMod, uint8_t tapMod, uint8_t rolloverMod, uint8_t keycode) {
```

Comment:
Just thinking out loud, might be a good idea to move this function into its own file under process_keycode/, and have it also handle `KC_SFTENT`.

Reply:
Possibly, thought the same thing. Will wait for more feedback from the contributors before making that switch.

Reply:
Yeah, actually, that would be best. 

Specifically, so we keep this consistent with the rest of the extended functionality.

Reply:
Moved to separate file & turned into a enable feature